### PR TITLE
feat:: V4 tokens tab, "Tokens" subpanel [3/n]

### PR DESCRIPTION
**Pull request by johnnyd-eth**
Screenshot 2024-07-19 at 7 21 42 pm

### CODE REVIEW

File: src/packages/v4/views/V4ProjectDashboard/V4ProjectTabs/V4TokensPanel/hooks/useV4TokensPanel.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  })
+  // const projectHasLegacyTokens = useProjectHasLegacyTokens()
+  const hasDeployErc20Permission = useV4WalletHasPermission(
+    V4OperatorPermission.DEPLOY_ERC20,
+  )
+  const projectHasErc20Token = Boolean(tokenAddress && !isZeroAddress(tokenAddress))
+
+  const { data: userTokenBalance, isLoading: userTokenBalanceLoading } =
+    useReadJbTokensTotalBalanceOf({
+      // address: controller.data, !!TODO
```
Comment (johnnyd-eth) [inline src/packages/v4/views/V4ProjectDashboard/V4ProjectTabs/V4TokensPanel/hooks/useV4TokensPanel.ts:35]: TODO: this should get user token balance working once we have this address

File: src/components/buttons/AddTokenToMetamaskButton.tsx [hunk 2]
```diff
@@ -49,8 +52,16 @@ function useAddTokenToWalletRequest() {
...
 }
 
-export function AddTokenToMetamaskButton({ className }: { className: string }) {
-  const addToken = useAddTokenToWalletRequest()
+export function AddTokenToMetamaskButton({ 
+  className,
+  tokenAddress
+}: { 
+  className: string,
+  tokenAddress: `0x${string}`
```
Comment (aeolianeth) [inline src/components/buttons/AddTokenToMetamaskButton.tsx:60]: Can use viem `Hash` type for these

File: src/packages/v2v3/components/V2V3Project/ProjectDashboard/components/TokensPanel/TokensPanel.tsx [hunk 3]
```diff
@@ -11,8 +11,8 @@ import { V2V3ClaimTokensModal } from 'packages/v2v3/components/V2V3Project/V2V3M
 import { V2V3MintModal } from 'packages/v2v3/components/V2V3Project/V2V3ManageTokensSection/AccountBalanceDescription/V2V3MintModal'
 import { useCallback, useState } from 'react'
 import { reloadWindow } from 'utils/windowUtils'
+import { AddTokenToMetamaskButton } from '../../../../../../../components/buttons/AddTokenToMetamaskButton'
```
Comment (aeolianeth) [inline src/packages/v2v3/components/V2V3Project/ProjectDashboard/components/TokensPanel/TokensPanel.tsx:14]: 🤔

File: src/packages/v4/hooks/useV4TotalTokenSupply.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { WeiPerEther } from '@ethersproject/constants';
+import { useJBContractContext, useReadJbControllerTotalTokenSupplyWithReservedTokensOf } from 'juice-sdk-react';
+import { useMemo } from 'react';
+
+export const useV4TotalTokenSupply = () => {
```
Comment (aeolianeth) [inline src/packages/v4/hooks/useV4TotalTokenSupply.ts:5]: How are you feeling about this approach to prefix everything with v4? Idk if its totally necessary given that all the v4 code is housed under `packages/v4`. No big deal, just a thought.

**Review by aeolianeth — APPROVED** [review]

File: src/packages/v4/hooks/useV4TotalTokenSupply.ts (hunk 4 above)
Comment (johnnyd-eth) [reply src/packages/v4/hooks/useV4TotalTokenSupply.ts:5]: Yep fair call, I'll move away from it.

**Pull request merged**